bagisto/bagisto
# Captcha Integrated #4632

## Description
Link: https://github.com/bagisto/bagisto/issues/4632

## For Developers
- Added two request i.e. `CustomerLoginRequest` and `CustomerRegistrationRequest` in `Customer` package.
- If the configuration is enabled in the backend then captcha response needs to be provided in the API also.
- `Captcha` alias is added. For rendering Captcha,

    ~~~php
    Captcha::render()
    ~~~

- For rendering scripts,

    ~~~php
    Captcha::renderJS()
    ~~~

- For backend validation, just pass `captcha` in request,

    ~~~php
   $this->validate(request(), [
       'g-recaptcha-response' => 'required|captcha',
    ]);
    ~~~

## How To Test This?
- Take a pull.
- Run `php artisan vendor:publish --all --force`.
- Run `php artisan optimize`.
- Configuration is given in the backend, just go to Configuration > Customer > Captcha.
- Set your API keys.
- Then check on the login and registration page and API also.
- In the API section, if the configuration is enabled, then the `g-captcha-repsonse` request is required.

## Documentation
- My pull request required some changes in the API docs.
